Using modulo with floating points causes panic

# Expected Behavior

I would expect that using floating points with the modulo operator is caught somewhere and an error is raised.  This is similar to what happens if you e.g. divide by zero:

```
jasper@kakigori$ opa run
OPA 0.10.5 (commit 8b08af7f, built at 2019-02-25T16:39:16Z)

Run 'help' to see a list of commands.

> 1 / 0
1 / 0: eval_internal_error: div: divide by zero
> 
```

# Actual Behavior

The process panics:

```
jasper@kakigori$ opa run
OPA 0.10.5 (commit 8b08af7f, built at 2019-02-25T16:39:16Z)

Run 'help' to see a list of commands.

> 1.1 % 1
panic: illegal value

goroutine 1 [running]:
github.com/open-policy-agent/opa/topdown/builtins.NumberToInt(0xc000203850, 0x3, 0xd6e360)
        /go/src/github.com/open-policy-agent/opa/topdown/builtins/builtins.go:163 +0x96
...
main.main()
        /go/src/github.com/open-policy-agent/opa/main.go:12 +0x2d
```

# Steps to Reproduce the Problem

  1. Start `opa run`
  2. Enter e.g. `1.1 % 1`

# Additional Info

- OPA version 0.10.5
- Linux 4.20.13 on x86_64

--- a/topdown/arithmetic.go
+++ b/topdown/arithmetic.go
@@ -123,7 +123,15 @@ func builtinRem(a, b ast.Value) (ast.Value, error) {
 	n2, ok2 := b.(ast.Number)
 
 	if ok1 && ok2 {
-		i, err := arithRem(builtins.NumberToInt(n1), builtins.NumberToInt(n2))
+
+		op1, err1 := builtins.NumberToInt(n1)
+		op2, err2 := builtins.NumberToInt(n2)
+
+		if err1 != nil || err2 != nil {
+			return nil, fmt.Errorf("modulo on floating-point number")
+		}
+
+		i, err := arithRem(op1, op2)
 		if err != nil {
 			return nil, err
 		}

--- a/topdown/builtins/builtins.go
+++ b/topdown/builtins/builtins.go
@@ -157,12 +157,13 @@ func FloatToNumber(f *big.Float) ast.Number {
 }
 
 // NumberToInt converts n to a big int.
-func NumberToInt(n ast.Number) *big.Int {
+// If n cannot be converted to an big int, an error is returned.
+func NumberToInt(n ast.Number) (*big.Int, error) {
 	r, ok := new(big.Int).SetString(string(n), 10)
 	if !ok {
-		panic("illegal value")
+		return nil, fmt.Errorf("illegal value")
 	}
-	return r
+	return r, nil
 }
 
 // IntToNumber converts i to a number.

--- a/topdown/topdown_test.go
+++ b/topdown/topdown_test.go
@@ -1049,6 +1049,7 @@ func TestTopDownArithmetic(t *testing.T) {
 		{"abs", []string{`p = true { abs(-10, x); x = 10 }`}, "true"},
 		{"remainder", []string{`p = x { x = 7 % 4 }`}, "3"},
 		{"remainder+error", []string{`p = x { x = 7 % 0 }`}, fmt.Errorf("modulo by zero")},
+		{"remainder+error+floating", []string{`p = x { x = 1.1 % 1 }`}, fmt.Errorf("modulo on floating-point number")},
 		{"arity 1 ref dest", []string{`p = true { abs(-4, a[3]) }`}, "true"},
 		{"arity 1 ref dest (2)", []string{`p = true { not abs(-5, a[3]) }`}, "true"},
 		{"arity 2 ref dest", []string{`p = true { a[2] = 1 + 2 }`}, "true"},

